File: lib/to_mdx/parsers/open_document/text/space.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Text
      class Space < Base
        def initialize(element = nil, context = {})
          super(element, context)
          count = element&.attributes&.[]("text:c")&.to_i || 1
          @text = " " * count
        end

        def empty?
          false
        end

        def to_mdx
          @text
        end

        def to_s
          @text
        end

        register_for "text:s"
      end
    end
  end
end

File: src/parsers/open-document/text/space.ts
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";

export class Space extends BaseElement {
  private text: string;

  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);
    this.text = " ".repeat(this.count);
  }

  empty(): boolean {
    return false;
  }

  toMdx(): string {
    return this.text;
  }

  toString(): string {
    return this.text;
  }

  get count(): number {
    return parseInt(this.attr("text:c") || "1", 10);
  }

  static {
    BaseElement.registerFor.call(this, "text:s");
  }
}
